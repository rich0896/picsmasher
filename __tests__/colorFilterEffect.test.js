const { ColorFilterEffect } = require('../effects.js');
const { createTestCanvas } = require('./helpers/testHelpers.js');

describe('ColorFilterEffect', () => {
    test('should apply color filter correctly', async () => {
        const { canvas, ctx } = await createTestCanvas();
        const effect = new ColorFilterEffect({
            red: 100,
            green: 50,
            blue: 0,
            intensity: 50,
        });

        const consoleSpy = jest
            .spyOn(console, 'log')
            .mockImplementation(() => {});

        effect.apply(ctx, canvas);

        expect(consoleSpy).toHaveBeenCalledWith('Applying Color Filter Effect');
        consoleSpy.mockRestore();

        const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
        expect(imageData).toMatchSnapshot();
    });
});
