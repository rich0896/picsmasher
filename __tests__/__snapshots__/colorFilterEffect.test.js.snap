// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorFilterEffect should apply color filter correctly 1`] = `ImageData { width: 200, height: 200, data: [100, 100, 100, 255, 100, 100, 100, 255, 100, 100...], length: 160000 }`;
